pyvisa_py/tcpip.py

```diff
@@ -41,6 +41,203 @@
 
 @Session.register(constants.InterfaceType.tcpip, "INSTR")
 class TCPIPInstrSession(Session):
+    """A class to dispatch to VXI11 or HiSLIP based on the protocol."""
+
+    def __new__(
+        cls,
+        resource_manager_session: VISARMSession,
+        resource_name: str,
+        parsed=None,
+        open_timeout: Optional[int] = None,
+    ):
+        obj: Union[TCPIPInstrHiSLIP, TCPIPInstrVxi11]
+
+        if parsed is None:
+            parsed = rname.parse_resource_name(resource_name)
+
+        if parsed.lan_device_name.lower().startswith("hislip"):
+            obj = TCPIPInstrHiSLIP(
+                resource_manager_session, resource_name, parsed, open_timeout
+            )
+        else:
+            obj = TCPIPInstrVxi11(
+                resource_manager_session, resource_name, parsed, open_timeout
+            )
+        return obj
+
+
+@Session.register(constants.InterfaceType.tcpip, "HISLIP")
+class TCPIPInstrHiSLIP(Session):
+    """A TCPIP Session built on socket standard library using HiSLIP protocol."""
+
+    # Override parsed to take into account the fact that this class is only used
+    # for a specific kind of resource
+    parsed: rname.TCPIPInstr
+
+    @staticmethod
+    def list_resources() -> List[str]:
+        # TODO: is there a way to get this?
```

Comment:
There is a pending PR for VXI-11, would something similar be envisioned for HiSLIP ?